### PR TITLE
Echidna tests

### PR DESCRIPTION
**Pull request by szotov**
Echidna tests for solidity contracts.
Two tests are broken because of error in `delegateCall` in echidna or one of it's dependencies (maybe `hevm`).
Example how to run: after echidna is installed run inside nucypher repo directory `echidna-test tests/blockchain/echidna/TokenTestSet.sol TokenTest1 --config="tests/blockchain/echidna/config.yaml"`

### CODE REVIEW

**Comment by KPrasch** [comment]
Just on a high-level can you describe the scope of this PR?

File: tests/blockchain/eth/contracts/main/policy_manager/test_policy_manager_operations.py [hunk 1]
```diff
@@ -153,9 +153,6 @@ def test_reward(testerchain, escrow, policy_manager):
 
 @pytest.mark.slow
 def test_refund(testerchain, escrow, policy_manager):
-    # Travel to the start of the next period to prevent problems with unexpected overflow first period
```
[the remark this answers is not in the document]
Comment (szotov) [reply tests/blockchain/eth/contracts/main/policy_manager/test_policy_manager_operations.py]: This lines were moved to `policy_manager/conftest.py`.
And it should fix this: 
https://circleci.com/gh/nucypher/nucypher/10271#tests/containers/2
(but difficult to prepare test)

**Comment by szotov** [comment]
> Just on a high-level can you describe the scope of this PR?

Echidna solidity tests without CI and one small fix for test.
Echidna tests:
* `config.yaml` - test configuration (use by default)
* `config-big-limit.yaml` - the same configuration except more test limit (for UserEscrow tests or maybe for all test in future)
* `Fixtures` - constants and factory methods
* `MasterContract` - helping contract to prepare main contracts together for test
* `*TestSet`, `*Test` - tests

**Comment by KPrasch** [comment]
Here are the steps I used to get a functional env:

```
$ docker pull trailofbits/echidna
$ docker run trailofbits/echidna
$ docker run -it -v "$PWD":/nucypher trailofbits/echidna
```

and to run the tests:

```
echidna-test /nucypher/tests/blockchain/echidna/TokenTestSet.sol TokenTest1 --config=/nucypher/tests/blockchain/echidna/config.yaml
```

At the moment tests fail to run due to a mismatched solidity compiler version:

```
/nucypher/tests/blockchain/echidna/TokenTestSet.sol:4:1: Error: Source file requires different compiler version (current compiler is 0.5.3+commit.10d17f24.Linux.g++ - note that nightly builds are considered to be strictly less than the released version
import "contracts/NuCypherToken.sol";
^----^
```

**Comment by szotov** [comment]
@KPrasch Change `RUN apt-get install -y solc` in Dockerfile to
```
RUN apt-get install -y wget
RUN wget "https://github.com/ethereum/solidity/releases/download/v0.4.25/solc-static-linux" -O /bin/solc
RUN chmod +x /bin/solc
```

**Comment by KPrasch** [comment]
Im still having a hard time running these tests  - I will try again soon.

**Comment by KPrasch** [comment]
@szotov - Still having the above Issue.  When you say to edit the `Dockerfile`, you are running a forked version of echidna?  Why does it not include a solidity compiler to begin with?

**Comment by szotov** [comment]
@KPrasch Yeah, I meant forked echidna.
In echidna repo was not suitable for us version of solc, so you can get latest master and just change in `Dockerfile` version of solidity.
Maybe problem because you run command not in /nucypher directory.

**Pull request closed without merging**